modules/quality_control/php/NDB_Menu_Filter_quality_control.class.inc

```diff
@@ -0,0 +1,471 @@
+                          "In MRI Violated Scans" => "Found in MRI Violated Scans",
+                          "Missing"               => "Missing",
+                         );
+        // Form Elements
+        $this->addSelect('site', 'Site', $siteList);
+        $this->addBasicText('candId', 'DCCID', ["size" => 9, "maxlength" => 7]);
+        $this->addBasicText('pSCID', 'PSCID', ["size" => 9, "maxlength" => 7]);
+        $this->addSelect('project', 'Project', $projectList2);
+        $this->addSelect('subproject', 'Subproject', $subprojectList2);
+        $this->addSelect('scanType', 'Scan Type', $scan_types);
+        $this->addSelect(
+            'mRIParameterForm',
+            "MRI Parameter Form",
+            array(
+             "Incomplete" => "Incomplete",
+             "Complete"   => "Complete",
+            )
+        );
+        $this->addSelect(
+            'scanDoneInMRIPF',
+            'Scan Done in MRI PF',
+            array(
+             "Yes" => "Yes",
+             "No"  => "No",
+            )
+        );
+        $this->addSelect(
+            'tarchive',
+            'Tarchive Status',
+            array(
+             "In DICOM" => "In DICOM Archives",
+             "Missing"  => "Missing",
+            )
+        );
+        $this->addSelect(
+            'scanLocation',
+            'Scan Location',
+            $scan_location
+        );
+        $this->addSelect(
+            'qCStatus',
+            'QC Status',
+            array(
+             "Pass" => "Pass",
+             "Fail" => "Fail",
+            )
+        );
+        $this->addSelect(
+            'uploadedBy',
+            'Uploaded By',
+            $uploaderList
+        );
+        $this->addSelect(
+            'selected',
+            'Selected',
+            array(
+             "True"  => "True",
+             "False" => "False",
+            )
+        );
+        return true;
+    }
+    /**
+     * Build a list of imaging QC to display in Data Table
+     *
+     * @return bool
+     * @throws DatabaseException
+     */
+    function _setupVariables()
+    {
+
+        $db =& Database::singleton();
+
+        $scan_types = $this->_getScanTypes();
+        foreach ($scan_types as $scan_type) {
+            $query_params        = array('scan_type' => $scan_type);
+            $acq_IDs[$scan_type] = $db->pselectOne(
+                "SELECT ID FROM mri_scan_type WHERE Scan_type=:scan_type",
+                $query_params
+            );
+            if (substr("$scan_type", 0, 2) == "t1") {
+                $scans_done[$scan_type] = "T1_Scan_done";
+            } elseif (substr("$scan_type", 0, 3) == "t2w") {
+                $scans_done[$scan_type] = "T2_Scan_done";
+            } elseif (substr("$scan_type", 0, 3) == "dti") {
+                $scans_done[$scan_type] = "dti_Scan_done";
+            } else {
+                $scans_done[$scan_type] = $scan_type . "_Scan_done";
```

Comment:
@llevitis 
I am not sure why it is necessary to hardcode t1, t2w and dti. This won't work across projects (for example, some projects might opt to call dti as dwi instead). Isn't the `else` part of those `if/elseif` statements the most generic way of doing what you are trying to accomplish?